src/dns_resolver_ares.c

```diff
@@ -126,36 +140,60 @@ static void query_completed_cb(void *arg, int status, int timeouts, struct hoste
             dns->is_complete = true;
             dns->in_progress = false;
         }
-        else
+#ifdef IPV6_ENABLED
+        else if (he->h_addrtype == AF_INET6)
         {
             ptr = dns->addrInfo;
-            
-            ptr->ai_addr = calloc(1, sizeof(struct sockaddr_in));
+
+            ptr->ai_addr = calloc(1, sizeof(struct sockaddr_in6));
+            addr6 = (void *)ptr->ai_addr;
+
             if(ptr->ai_addr == NULL)
             {
                 LogError("dns addrinfo ai_addr: allocation failed");
                 free(dns->addrInfo);
                 dns->is_failed = true;
                 dns->is_complete = true;
                 dns->in_progress = false;
+            } else {
+                memcpy(&addr6->sin6_addr, he->h_addr_list[0], sizeof(struct in6_addr));
+                addr6->sin6_family = he->h_addrtype;
+                addr6->sin6_port = htons((unsigned short)dns->port);
+
+                /* Codes_SRS_dns_resolver_30_033: [ If dns_resolver_is_create_complete has returned true and the lookup process has failed, dns_resolver_get_ipv4 shall return 0. ]*/
+                memcpy(dns->ip_v6, EXTRACT_IPV6(ptr), 16); // IPv6 address is 16 bytes
+                dns->addrInfo->ai_family = AF_INET6;
+                dns->is_failed = (dns->ip_v6 == 0);
+                dns->is_complete = true;
+                dns->in_progress = false;
             }
-            else
-            {
-                addr = (void *)ptr->ai_addr;
+        }
+#endif // IPV6_ENABLED
+        else
+        {
+            ptr = dns->addrInfo;
 
-                if (he->h_addrtype == AF_INET)
```

Comment:
Should this if statement still be in the IPv4 logic?

Reply:
Good catch! Added it back